src/main/java/Server/MopeLSPServerLauncher.java

```diff
@@ -29,20 +29,29 @@
     private static Logger logger = LoggerFactory.getLogger(MopeLSPServerLauncher.class);
     private static ConfigObject configObject;
 
+    static {
+        serverSocket = null;
+        server = null;
+        configObject = null;
+    }
+
     public MopeLSPServerLauncher() throws IOException {
         configObject = new ConfigObject();
         readConfig();
         server = new MopeLSPServer(configObject);
         serverSocket = new ServerSocket(configObject.port);
     }
+    public static void setExecutorPool() {
+        executor = Executors.newCachedThreadPool();
+    }
 
     public void launchServer() {
 
         System.setProperty(Log4jLoggerAdapter.ROOT_LOGGER_NAME, "TRACE");
 
         logger.info("Server socket listening on port " + configObject.port );
         System.out.flush();
-        executor = Executors.newCachedThreadPool();
+        //executor = Executors.newCachedThreadPool();
```

Comment:
Also, creating the executor in `main()` instead of `launchServer()` means that the tests now fail with a null pointer exception, since the `ServerIntegrationTest` calls `launchServer()` but not `main()`.

Reply:
Well this is probably because i was not into the project for weeks and do not know every change since then, but yesterday I already had the feeling, theres something wrong with these changes. I will appreciate it.